From 05093ae00cd2163e1f1b8edb22e2c4387e5b70dd Mon Sep 17 00:00:00 2001
From: Igal Liberman <igall@marvell.com>
Date: Sun, 20 Aug 2017 17:11:30 +0300
Subject: [PATCH 1182/1345] dts: a80x0: a70x0: move eip197 enablement to SoC
 level

commit  a23e5ad39a53947f591793736871dbb2454d9888 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

EIP197 isn't board dependent so we need to enable it
in the SoC level.

Change-Id: Iad920e3c8fcedfccb74f178a35482540672b10f2
Signed-off-by: Igal Liberman <igall@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/43236
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-7040-db-B.dts   |    3 ---
 arch/arm64/boot/dts/marvell/armada-7040-db-C.dts   |    3 ---
 arch/arm64/boot/dts/marvell/armada-7040-db-D.dts   |    3 ---
 arch/arm64/boot/dts/marvell/armada-7040-db-E.dts   |    3 ---
 .../boot/dts/marvell/armada-7040-db-all-pcie.dts   |    3 ---
 .../arm64/boot/dts/marvell/armada-7040-db-nand.dts |    3 ---
 .../dts/marvell/armada-7040-db-pp2-vanilla.dts     |    3 ---
 arch/arm64/boot/dts/marvell/armada-7040-pcac.dts   |    3 ---
 arch/arm64/boot/dts/marvell/armada-7040-pd.dts     |    3 ---
 .../boot/dts/marvell/armada-70x0-customer.dts      |    3 ---
 arch/arm64/boot/dts/marvell/armada-8040-db-B.dts   |    6 ------
 arch/arm64/boot/dts/marvell/armada-8040-db-C.dts   |    6 ------
 arch/arm64/boot/dts/marvell/armada-8040-db-D.dts   |    6 ------
 arch/arm64/boot/dts/marvell/armada-8040-db-E.dts   |    6 ------
 .../dts/marvell/armada-8040-db-pp2-vanilla.dts     |    6 ------
 .../dts/marvell/armada-8040-mcbin-pp2-vanilla.dts  |    6 ------
 arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts  |    6 ------
 .../boot/dts/marvell/armada-80x0-customer.dts      |    9 ---------
 arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi    |    2 +-
 arch/arm64/boot/dts/marvell/armada-cp110.dtsi      |    2 +-
 20 files changed, 2 insertions(+), 83 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db-B.dts b/arch/arm64/boot/dts/marvell/armada-7040-db-B.dts
index ac1417a..d35b835 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db-B.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db-B.dts
@@ -160,9 +160,6 @@
 					status = "okay";
 				};
 			};
-			eip197: eip197@800000 {
-				status = "okay";
-			};
 			comphy {
 				status = "okay";
 			};
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db-C.dts b/arch/arm64/boot/dts/marvell/armada-7040-db-C.dts
index f16d47b..6a28d20 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db-C.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db-C.dts
@@ -179,9 +179,6 @@
 					status = "okay";
 				};
 			};
-			eip197: eip197@800000 {
-				status = "okay";
-			};
 			comphy {
 				status = "okay";
 			};
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db-D.dts b/arch/arm64/boot/dts/marvell/armada-7040-db-D.dts
index 8a8e94a..2355161 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db-D.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db-D.dts
@@ -186,9 +186,6 @@
 					status = "okay";
 				};
 			};
-			eip197: eip197@800000 {
-				status = "okay";
-			};
 			comphy {
 				status = "okay";
 			};
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db-E.dts b/arch/arm64/boot/dts/marvell/armada-7040-db-E.dts
index 16381e1..7e223f5 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db-E.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db-E.dts
@@ -158,9 +158,6 @@
 					status = "disabled";
 				};
 			};
-			eip197: eip197@800000 {
-				status = "okay";
-			};
 			comphy {
 				status = "okay";
 			};
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db-all-pcie.dts b/arch/arm64/boot/dts/marvell/armada-7040-db-all-pcie.dts
index f1db8b0..83d7f4f 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db-all-pcie.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db-all-pcie.dts
@@ -149,9 +149,6 @@
 					status = "okay";
 				};
 			};
-			eip197: eip197@800000 {
-				status = "okay";
-			};
 			comphy {
 				status = "okay";
 			};
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db-nand.dts b/arch/arm64/boot/dts/marvell/armada-7040-db-nand.dts
index 55a0545..319b7f3 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db-nand.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db-nand.dts
@@ -160,9 +160,6 @@
 					status = "okay";
 				};
 			};
-			eip197: eip197@800000 {
-				status = "okay";
-			};
 			comphy {
 				status = "okay";
 			};
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db-pp2-vanilla.dts b/arch/arm64/boot/dts/marvell/armada-7040-db-pp2-vanilla.dts
index 3462379..046b131 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db-pp2-vanilla.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db-pp2-vanilla.dts
@@ -195,9 +195,6 @@
 					status = "okay";
 				};
 			};
-			eip197: eip197@800000 {
-				status = "okay";
-			};
 			comphy {
 				status = "disabled";
 			};
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-pcac.dts b/arch/arm64/boot/dts/marvell/armada-7040-pcac.dts
index 5e4e34d..91c6d7b 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-pcac.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-pcac.dts
@@ -207,9 +207,6 @@
 					status = "disabled";
 				};
 			};
-			eip197: eip197@800000 {
-				status = "okay";
-			};
 		};
 
 		pcie-ep@0x600000 {
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-pd.dts b/arch/arm64/boot/dts/marvell/armada-7040-pd.dts
index b8f2c71..7154484 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-pd.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-pd.dts
@@ -253,9 +253,6 @@
 					status = "okay";
 				};
 			};
-			eip197: eip197@800000 {
-				status = "okay";
-			};
 		};
 
 		pcie@0x640000 {
diff --git a/arch/arm64/boot/dts/marvell/armada-70x0-customer.dts b/arch/arm64/boot/dts/marvell/armada-70x0-customer.dts
index c6fde62..9c1c150 100644
--- a/arch/arm64/boot/dts/marvell/armada-70x0-customer.dts
+++ b/arch/arm64/boot/dts/marvell/armada-70x0-customer.dts
@@ -281,9 +281,6 @@
 					status = "disabled";
 				};
 			};
-			eip197: eip197@800000 {
-				status = "disabled";
-			};
 			comphy {
 				status = "disabled";
 			};
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-B.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-B.dts
index 264dc82..3478e09 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-B.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-B.dts
@@ -132,9 +132,6 @@
 					status = "okay";
 				};
 			};
-			eip197@800000 {
-				status = "okay";
-			};
 			comphy {
 				status = "okay";
 			};
@@ -216,9 +213,6 @@
 				pinctrl-names = "default";
 				pinctrl-0 = <&cps_spi1_pins>;
 			};
-			eip197@800000 {
-				status = "okay";
-			};
 		};
 		pcie@0x600000 {
 			status = "okay";
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-C.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-C.dts
index e32dca0..56b849a 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-C.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-C.dts
@@ -147,9 +147,6 @@
 					status = "okay";
 				};
 			};
-			eip197@800000 {
-				status = "okay";
-			};
 			comphy {
 				status = "okay";
 			};
@@ -242,9 +239,6 @@
 				pclk-freq-mhz = <8>;
 				status = "okay";
 			};
-			eip197@800000 {
-				status = "okay";
-			};
 		};
 		pcie@0x600000 {
 			status = "okay";
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-D.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-D.dts
index a02dba9..3016cf4 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-D.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-D.dts
@@ -148,9 +148,6 @@
 					status = "okay";
 				};
 			};
-			eip197@800000 {
-				status = "okay";
-			};
 			comphy {
 				status = "okay";
 			};
@@ -216,9 +213,6 @@
 					status = "disabled";
 				};
 			};
-			eip197@800000 {
-				status = "disabled";
-			};
 			comphy {
 				status = "okay";
 			};
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-E.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-E.dts
index 11bef47..f4b5592 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-E.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-E.dts
@@ -150,9 +150,6 @@
 					status = "disabled";
 				};
 			};
-			eip197@800000 {
-				status = "okay";
-			};
 			comphy {
 				status = "okay";
 			};
@@ -234,9 +231,6 @@
 				pinctrl-names = "default";
 				pinctrl-0 = <&cps_spi1_pins>;
 			};
-			eip197@800000 {
-				status = "okay";
-			};
 		};
 		pcie@0x600000 {
 			status = "okay";
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-pp2-vanilla.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-pp2-vanilla.dts
index 3731296..a6ee25b 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-pp2-vanilla.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-pp2-vanilla.dts
@@ -173,9 +173,6 @@
 					status = "okay";
 				};
 			};
-			eip197@800000 {
-				status = "okay";
-			};
 			comphy {
 				status = "disabled";
 			};
@@ -264,9 +261,6 @@
 					status = "disabled";
 				};
 			};
-			eip197@800000 {
-				status = "okay";
-			};
 			comphy {
 				status = "disabled";
 			};
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-mcbin-pp2-vanilla.dts b/arch/arm64/boot/dts/marvell/armada-8040-mcbin-pp2-vanilla.dts
index 820e5b6..f3021d2 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-mcbin-pp2-vanilla.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-mcbin-pp2-vanilla.dts
@@ -202,9 +202,6 @@
 					status = "disabled";
 				};
 			};
-			eip197@800000 {
-				status = "okay";
-			};
 		};
 		pcie@0x600000 {
 			num-lanes = <4>;
@@ -323,9 +320,6 @@
 					status = "disabled";
 				};
 			};
-			eip197@800000 {
-				status = "okay";
-			};
 		};
 	};
 };
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts b/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts
index 05c6491..9dda497 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts
@@ -119,9 +119,6 @@
 					status = "okay";
 				};
 			};
-			eip197@800000 {
-				status = "okay";
-			};
 		};
 		pcie@0x600000 {
 			num-lanes = <4>;
@@ -184,9 +181,6 @@
 					status = "okay";
 				};
 			};
-			eip197@800000 {
-				status = "okay";
-			};
 		};
 	};
 
diff --git a/arch/arm64/boot/dts/marvell/armada-80x0-customer.dts b/arch/arm64/boot/dts/marvell/armada-80x0-customer.dts
index 608cb91..7f783eb 100644
--- a/arch/arm64/boot/dts/marvell/armada-80x0-customer.dts
+++ b/arch/arm64/boot/dts/marvell/armada-80x0-customer.dts
@@ -201,9 +201,6 @@
 					status = "disabled";
 				};
 			};
-			eip197@800000 {
-				status = "disabled";
-			};
 			comphy {
 				status = "disabled";
 			};
@@ -214,9 +211,6 @@
 					marvell,function = "gpio";
 				};
 			};
-			eip197@800000 {
-				status = "okay";
-			};
 		};
 
 		pcie@0x640000 {
@@ -317,9 +311,6 @@
 				pclk-freq-mhz = <8>;
 				status = "disabled";
 			};
-			eip197@800000 {
-				status = "okay";
-			};
 		};
 		pcie@0x640000 {
 			status = "disabled";
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
index cb3309c..0831c18 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
@@ -368,7 +368,7 @@ cps_eip197: eip197@800000 {
 			  "ring2", "ring3", "eip_out";
 	clocks = <&cps_syscon0 1 26>;
 	cell-index = <1>;
-	status = "disabled";
+	status = "okay";
 };
 
 comphy1: comphy {
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
index 1d415d8..42e800a 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
@@ -466,7 +466,7 @@ cpm_eip197: eip197@800000 {
 			  "ring2", "ring3", "eip_out";
 	clocks = <&cpm_syscon0 1 26>;
 	cell-index = <0>;
-	status = "disabled";
+	status = "okay";
 };
 
 comphy0: comphy {
-- 
1.7.9.5

